src/Npgsql/NpgsqlWriteBuffer.cs

```diff
@@ -377,6 +377,44 @@ public void WriteBytes(byte[] buf, int offset, int count)
             _writePosition += count;
         }
 
+        public Task WriteBytesRaw(byte[] bytes, bool async)
+        {
+            if (bytes.Length <= WriteSpaceLeft)
+            {
+                WriteBytes(bytes);
+                return PGUtil.CompletedTask;
+            }
+            else
+            {
+                return WriteBytesLong(bytes, async);
+            }
+        }
+
+        async Task WriteBytesLong(byte[] bytes, bool async)
```

Comment:
Make it a local function in `WriteBytesRaw` and remove `else`. You can change the condition and swap bodies of `if` and `else` like for `INpgsqlTypeHandler<byte[]>.Read`.